File: Python/military_time.py
import os
os.system("clear")

time = input()
time = time.upper()
time = time.split(":")

if "PM" in time[1] and int(time[0]) <= 11:
  H = int(time[0]) + 12
  HH = str(H) + ":" + str(time[1][0]) + str(time[1][1])
  print(HH)
elif int(time[0]) == 12  and "AM" in time[1]:
  H = '00'
  HH = str(H) + ":" + str(time[1][0]) + str(time[1][1])
  print(HH)
elif int(time[0]) <= 12  and "AM" in time[1]:
  if int(time[0]) <= 9:
    H = "0" + str(int(time[0]))
  else:
    H = time[0]
  HH = str(H) + ":" + str(time[1][0]) + str(time[1][1])
  print(HH)
elif int(time[0]) == 12  and "PM" in time[1]:
  H = time[0]
  HH = str(H) + ":" + str(time[1][0]) + str(time[1][1])
  print(HH)

File: Ruby/militarytime.rb
time = gets.chomp.upcase.split(':')
if time[1].include? 'PM' and time[0].to_i <= 11
  H = time[0].to_i + 12
  HH = "#{H}:#{time[1][0..1]}"
  print HH
elsif time[0].to_i == 12  and time[1].include? 'AM'
  H = '00'
  HH = "#{H}:#{time[1][0..1]}"
  print(HH)
elsif time[0].to_i <= 12  and time[1].include? 'AM'
  if time[0].to_i <= 9
    H = "0#{time[0]}"
  else
    H = time[0]
  end
  HH = "#{H}:#{time[1][0..1]}"
  print(HH)
elsif time[0].to_i == 12  and time[1].include? 'PM'
  H = time[0]
  HH = "#{H}:#{time[1][0..1]}"
  print(HH)
end
